delocate/fuse.py

```diff
@@ -83,24 +124,46 @@ def fuse_trees(to_tree, from_tree, lib_exts=(".so", ".dylib", ".a")):
                 _copyfile(from_path, to_path)
 
 
-def fuse_wheels(to_wheel, from_wheel, out_wheel):
+def fuse_wheels(
+    to_wheel: str | PathLike,
+    from_wheel: str | PathLike,
+    out_wheel: str | PathLike | None = None,
+) -> Path:
     """Fuse `from_wheel` into `to_wheel`, write to `out_wheel`.
 
     Parameters
     ----------
-    to_wheel : str
-        filename of wheel to fuse into
-    from_wheel : str
-        filename of wheel to fuse from
-    out_wheel : str
-        filename of new wheel from fusion of `to_wheel` and `from_wheel`
+    to_wheel : str or Path-like
+        The path of the wheel to fuse into.
+    from_wheel : str or Path-like
+        The path of the wheel to fuse from.
+    out_wheel : str or Path-like, optional
+        The path of the new wheel from fusion of `to_wheel` and `from_wheel`. If
+        a full path is given, (including the filename) it will be used as is. If
+        a directory is given, the fused wheel will be stored in the directory,
+        with the name of the wheel automatically determined. If no path is
+        given, the fused wheel will be stored in the same directory as
+        `to_wheel`, with the name of the wheel automatically determined.
+
+    Returns
+    -------
+    out_wheel : Path
+        The path of the new wheel from fusion of `to_wheel` and `from_wheel`.
     """
-    to_wheel, from_wheel, out_wheel = [
-        abspath(w) for w in (to_wheel, from_wheel, out_wheel)
-    ]
+    to_wheel, from_wheel = [Path(w).resolve() for w in (to_wheel, from_wheel)]
```

Comment:
Add `strict=True` to `Path.resolve` to validate that input paths exist.
```suggestion
    to_wheel, from_wheel = [Path(w).resolve(strict=True) for w in (to_wheel, from_wheel)]
```